R/teal_data-get_code.R

```diff
@@ -103,23 +105,23 @@
 #' @aliases get_code,teal_data-method
 #'
 #' @export
-setMethod("get_code", signature = "teal_data", definition = function(object, deparse = TRUE, datanames = NULL, ...) {
-  checkmate::assert_character(datanames, min.len = 1L, null.ok = TRUE)
-  checkmate::assert_flag(deparse)
-
-  code <- if (!is.null(datanames)) {
-    get_code_dependency(object@code, datanames, ...)
-  } else {
-    object@code
-  }
+setMethod("get_code",
+  signature = "teal_data",
+  definition = function(object, deparse = TRUE, datanames = NULL, names = datanames, ...) {
+    checkmate::assert_character(datanames, min.len = 1L, null.ok = TRUE)
+    checkmate::assert_character(names, min.len = 1L, null.ok = TRUE)
+    checkmate::assert_flag(deparse)
 
-  if (deparse) {
-    if (length(code) == 0) {
-      code
-    } else {
-      paste(code, collapse = "\n")
+    if (!is.null(datanames)) {
+      lifecycle::deprecate_warn(
+        when = "0.6.1",
+        what = "get_code()",
+        with = "teal.code::get_code()",
+        always = TRUE
+      )
     }
-  } else {
-    parse(text = paste(c("{", code, "}"), collapse = "\n"), keep.source = TRUE)
+    class(object) <- "qenv"
```

Comment:
I am not sure this is the cleanest way, but we need a way to call `teal.code::get_code` directly for the `qenv` object. Without that this calls the generic, that uses some `pdf` devices and crashes, and does not extract the code at all.

Reply:
I guess `CallNextMethod()` is the answer here.

Reply:
Thanks! Nice finding - incorporated in here 7d1c6eebffd989cef1bea7129a1cd2955059bdc5